Automattic/woocommerce-payments
# Remove legacy UPE and card front-end scripts and the corresponding code for script registration

Closes https://github.com/Automattic/woocommerce-payments/issues/7780

This PR removes checkout scripts for shortcode and Blocks checkout that were handling the legacy UPE/split UPE and legacy card flows. The below list contains changes made with some testing scenarios outlined.

#### Changes proposed in this Pull Request
* [remove legacy card checkout script and its dependencies](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/7dfed63a51954a3cc602c13ca9df8c6d3a0ff19e)
    * performed a side-by-side comparison looking for any discrepancies between the new solution and the removed one
    * found three areas that need a slight improvement, which are not blocking this PR and mainly cover other UPEs/legacy cards as well
        *  https://github.com/Automattic/woocommerce-payments/issues/7788
        * https://github.com/Automattic/woocommerce-payments/issues/7799
        * p1701097214386029-slack-C01B8KNUYSW 
    * no manual tests needed since the file was removed which wasn't rolled out
* [cleanup shortcode checkout styles and use them in deferred UPE](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/9ea9f885025eb6c18f9c50a148f9e105b1700d11)
    * cleaned up `styles.scss` by removing selectors not applicable to the deferred UPE elements
    * start applying styles to the deferred UPE (it wasn't the case before). To check this
        *  🧪 checkout `develop`, enable Stripe Link, navigate to the shortcode checkout, and confirm [the button placement is misaligned](https://d.pr/i/IrExyA)
        * 🧪 checkout `cleanup/legacy-upe-and-card-front-end`, `npm run build`, refresh the page and confirm the button is now placed correctly
    * 🧪 confirm that credit card and any other non-card payment are displayed correctly on the checkout
* [remove split upe checkout script and its dependencies](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/b1e8044e90ec53dea08f4c07acf788f38140c040)
    * performed a side-by-side comparison looking for any discrepancies between the new solution and the removed one
    * no manual tests needed since the file was removed which wasn't rolled out
* [remove legacy upe checkout script and its dependencies](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/cad8cf7a836079472fd92bd7c9296a0624329255)
    * same as previously: performed a side-by-side comparison looking for any discrepancies between the new solution and the removed one
    * no manual tests needed
* [slightly re-organize tests by moving them to a proper package](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/9cb71061e0469c34908b6ca279c474828db30045)
    * move upe utils test coverage to the proper file
* misc [simplify path to the checkout scripts](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/db12cba1f7b36a4a79ed32c870f325393973876d)
* [remove old blocks scripts and ensure remaining implementation has everything needed](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/c3153db48e466f92fa9f0f7160ce74ca54b58ac9)
    * 🧪 confirm that WooPay works on blocks checkout both through express checkout button and email input field
        *  styles should be applied correctly, [not like here](https://github.com/Automattic/woocommerce-payments/assets/22319444/8b860a74-de81-4672-a316-5ce644ac4017)
* [add `fontRules` for current elements](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/f203c09fdfc6670a9bc50d699b7b5bafb0bef785). This was added to stay in line with previous implementations.
  <details>
    <summary>Before & After</summary>

    | Before | After |
    |--------|-------|
    | <img width="448" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/22319444/e95aa5a1-5a67-4c97-94b0-8dc357ab05f0"> | <img width="447" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/22319444/473d6c8b-dd37-40a7-9ddf-bb0df993ad49"> |

  </details>
* [remove legacy upe blocks front-end implementation](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/03e458af5153b849fa245111aa220cc61f5db34c)
* [remove upe-split blocks checkout scripts](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/3a684ce266bfd1a17a69246ed801ac60ddb3022d)
* [remove legacy card checkout scripts and simplify the naming of existing](https://github.com/Automattic/woocommerce-payments/pull/7795/commits/37ec20299fdd221bfb3d3e17965321199f4e5697)
    * performed comparison between the removed checkout and deferred UPE










#### Testing instructions
Besides the scenarios outlined above, please make sure to sanity check the payment processing for the deferred UPE with: 
* CC
* any non-CC method e.g. giropay
* Stripe Link
* WooPay (both express checkout & through email input field)





-------------------


**Post merge**
